ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -4091,10 +4091,32 @@ let vm_clone_aux clone_op cloned_string printer include_template_vms rpc
        (List.map (fun vm -> Client.VM.get_uuid rpc session_id vm) new_vms)
     )
 
+let vm_snap_aux clone_op cloned_string printer include_template_vms rpc
+    session_id params =
+  let new_name = List.assoc "new-name-label" params in
+  let desc =
+    try Some (List.assoc "new-name-description" params) with _ -> None
+  in
+  let new_vms =
+    do_vm_op printer ~include_template_vms rpc session_id
+      (fun vm -> clone_op ~rpc ~session_id ~vm:(vm.getref ()) ~new_name ~ignored_vdis:[])
+      params
+      ["new-name-label"; "new-name-description"]
+  in
+  Option.iter
+    (fun desc ->
+      Client.VM.set_name_description rpc session_id (List.hd new_vms) desc
+      )
+    desc ;
+  printer
+    (Cli_printer.PList
+       (List.map (fun vm -> Client.VM.get_uuid rpc session_id vm) new_vms)
+    )
+
```

Comment:
I've duplicated this piece of code to work around the compilation issue and keep going.
But I think there's a better solutiion that I don't know about.